test/blas_test.hpp

```diff
@@ -204,8 +204,13 @@ class BLAS_Test<blas_test_args<ScalarT_, MetadataT_, ExecutorType_>>
   }
 };
 
-#define ASSERT_T_EQUAL(T, val1, val2)                                         \
-  ASSERT_PRED_FORMAT2(::testing::internal::CmpHelperFloatingPointEQ<T>, val1, \
-                      val2)
+constexpr const int ulp_required_in_assert = 20;
+
+#define ASSERT_T_EQUAL(T, val1, val2)                                   \
+  do {                                                                  \
+    T precision = std::numeric_limits<T>::epsilon() *                   \
+                  std::abs((val1) + (val2)) * ::ulp_required_in_assert; \
```

Comment:
Shouldn't it be `fabs`?

Reply:
Also if you can put a link or explanation of where that 20 comes from that would be nice

Reply:
std::abs and std::fabs do the same thing if cmath is imported. Because C++.

Reply:
Are you absolutely sure about this? I don't see floating point versions there: https://en.cppreference.com/w/cpp/numeric/math/abs